### PR TITLE
feat: add casl packages to the stack to get a right policy via hapify

### PR DESCRIPTION
**Pull request by floross**
- fix: throw a more explicit exception in case something is not going good with the authentication
- feat: add nestjs casl and template casl to the stack
- feat: wip add cals ability to the stack

### CODE REVIEW

File: apps/pwa/src/assets/app-config.json [hunk 1]
```diff
@@ -0,0 +1 @@
+{"NODE_ENV":"development","PORT":"3000","API_URI":"http://localhost:4200/api","POSTGRES_USER":"postgresuser","POSTGRES_PASSWORD":"koh1Aiphen5feesuophie7aiNg0Yeep5","POSTGRES_DB":"stack","POSTGRES_HOST":"localhost","POSTGRES_PORT":"5432","POSTGRES_SCHEMA":"public","TRACTR_DATABASE_URL":"postgresql://${POSTGRES_USER}:${POSTGRES_PASSWORD}@${POSTGRES_HOST}:${POSTGRES_PORT}/${POSTGRES_DB}?schema=${POSTGRES_SCHEMA}${POSTGRES_OPTIONS}"}
```
Comment (maxmousse) [inline apps/pwa/src/assets/app-config.json]: Do we want to versionate this file ?

File: libs/angular/authentication/src/lib/services/session.service.ts [hunk 2]
```diff
@@ -62,6 +63,11 @@ export function SessionServiceFactory<
         if (err && err.status === 401) return of(null);
         throw err;
       }),
+      map((mockUser) => {
```
Comment (maxmousse) [inline libs/angular/authentication/src/lib/services/session.service.ts]: Is this code for developement purpose ?

File: libs/angular/casl/src/lib/services/angular-casl-update-abilities.service.ts [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+} from '@tractr/angular-authentication';
+import { Unsubscriber } from '@tractr/angular-tools';
+import { CaslUserRoles } from '@tractr/common';
+
+@Injectable()
+export class CaslUpdateAbilitiesService extends Unsubscriber {
+  constructor(
+    @Inject(CASL_MODULE_OPTIONS)
+    private caslOptions: CaslOptions,
+    @Inject(SESSION_SERVICE)
```
Comment (maxmousse) [inline libs/angular/casl/src/lib/services/angular-casl-update-abilities.service.ts:20]: I don't understand how session service is available as it is not provided in the module

**Review by maxmousse — APPROVED** [review]

File: libs/angular/casl/src/lib/services/angular-casl-update-abilities.service.ts (hunk 3 above)
Comment (floross) [reply libs/angular/casl/src/lib/services/angular-casl-update-abilities.service.ts:20]: Angular put the service in the root injector and it's available in all other modules

**Pull request merged**